include/openthread/thread.h

```diff
@@ -286,6 +286,18 @@ otError otThreadSetJoinerAdvertisement(otInstance *   aInstance,
                                        const uint8_t *aAdvData,
                                        uint8_t        aAdvDataLength);
 
+/**
+ * This method returns the Thread Joiner Advertisement.
+ *
+ * @param[in]   aInstance       A pointer to an OpenThread instance.
+ * @param[out]  aOui            The Vendor IEEE OUI value that will be included in the Joiner Advertisement.
+ * @param[out]  aAdvDataLength  The length of Advertisement data in bytes.
+ *
+ * @retval A pointer to the Advertisement data, or NULL if none is set.
+ *
+ */
+const uint8_t *otThreadGetJoinerAdvertisement(otInstance *aInstance, uint32_t *aOui, uint8_t *aAdvDataLength);
```

Comment:
Could you help explain what's the use case of these "Get" APIs and why do we need them?

Reply:
A `getter` usually comes with a `setter` (It is common to have `write-only` API, but is unusual to have `read-only` API). For debugging, the getter is helpful to validate that we have the advertisement data set as expected.

Reply:
If only for debugging/testing purpose, we can put this in the `REFERENCE DEVICE` scope. 
Thoughts? @wgtdkp @librasungirl @lanyuwen

Reply:
@simonlingoogle  I think the user of OpenThread have the needs to verify that their product have correct advertisement data set.

Reply:
> @simonlingoogle I think the user of OpenThread have the needs of verifying that their products have correct advertisement data set.

We can have the API always available and CLI commands reference device only.

Reply:
@lanyuwen @simonlingoogle Decided to remove the setter as it seems not critical for now.